lib/logger/lib/logger/formatter.ex

```diff
@@ -62,20 +62,28 @@ defmodule Logger.Formatter do
   defp prune_binary(<<_, t::binary>>, acc), do: prune_binary(t, <<acc::binary, @replacement>>)
   defp prune_binary(<<>>, acc), do: acc
 
-  @doc ~S"""
-  Compiles a format string into a data structure that the `format/5` can handle.
+  @doc """
+  Compiles a format string into a data structure that `format/5` can handle.
+
+  Check the module doc for documentation on the valid parameters that
+  will be interpolated in the pattern. If you pass `nil` as the pattern,
+  the pattern defaults to:
+
+      #{inspect(@default_pattern)}
```

Comment:
Doing this requires us to drop the `~S` sigil and use `\\n` below, but I think it's worth it so that we don't repeat the pattern here.